sycl/plugins/unified_runtime/ur/adapters/cuda/usm.cpp

```diff
@@ -42,13 +36,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMHostAlloc(
     Result = Err;
   }
 
-  UR_ASSERT(!pUSMDesc || (pUSMDesc->align == 0 ||
-                          ((pUSMDesc->align & (pUSMDesc->align - 1)) == 0)),
-            UR_RESULT_ERROR_INVALID_VALUE);
-
-  assert(Result == UR_RESULT_SUCCESS &&
-         (!pUSMDesc || pUSMDesc->align == 0 ||
-          reinterpret_cast<std::uintptr_t>(*ppMem) % pUSMDesc->align == 0));
+  if (Result == UR_RESULT_SUCCESS) {
```

Comment:
This slightly changes the behaviour of `urUSMHostAlloc`, previously the function asserted on the result (exception state of `cuMemAllocHost`), now it will return it.

Reply:
This assert was a little different before the port:
```
  assert(alignment == 0 ||
         (result == PI_SUCCESS &&
          reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0));
```

@callumfare can weigh in on this but I think the intent was more to check the alignment if the allocation is successful rather than assert if it's not.

And since this patch is removing the size check to rely on the CUDA call to handle too large sizes, it means we now need to propagate the error code from the CUDA call.

Reply:
Yeah this makes sense to me, without this change the invalid allocation size error won't propagate back up to the user, but the behavior of the alignment check is unchanged